tdd/assignment5.test.js

```diff
@@ -10,13 +10,15 @@ let transaction_open = false;
 const lines = [];
 
 const one_sql_line = () => {
-  while (true) {
-    let line = lines.shift();
-    if (line == undefined) return line;
-    line = line.trim();
-    if (line.length === 0 || line[0] === "#") continue; // skip comments and blank lines
-    return line;
+  let statement = "";
+
+  while (lines.length > 0) {
+    const line = lines.shift().trim();
+    if (line.length === 0 || line.startsWith("#")) continue;
+    statement += " " + line;
+    if (line.includes(";")) break; // If a semicolon is found, the statement is complete
```

Comment:
It might be better to do line.endswith(";")  instead of line.includes(";").  I don't think we have any examples of this in the homework, but an SQL statement could have a semicolon in one of the strings, like SELECT * FROM table_x WHERE name = 'George Jones; esquire' followed by more SQL on a subsequent line.  The endswith() would handle that case.  Just a nit.